src/main.py

```diff
@@ -121,10 +126,13 @@ def main():
         else:
             logger.info(f"Ignoring test: {test.name()}")
 
+    # cleanup any namespaces on the drive
```

Comment:
Do the previous tests need to be wrapped in a try/except block, and then the restore drive be in the except block as well?  

We kind of need to make sure it's run no matter if the tests passed or failed.  And I think tests should be assumed to be able to throw a failure.

Reply:
Agreed, the drive should be restored irrespective of whether the tests passed or failed. We do not want to run the restore after every test though so we can handle the exception and run the restore after the tests.